### PR TITLE
Delete comments

### PR DESCRIPTION
**Pull request by dimak1**
@dleard this a new PR for this feature since the other PR was failing on CI e2e test (due to branch sitting in my fork repo).

### CODE REVIEW

File: app/containers/Applications/ApplicationCommentsByForm.tsx [hunk 1]
```diff
@@ -46,6 +73,9 @@ export const ApplicationCommentsByForm: React.FunctionComponent<Props> = props =
               type="checkbox"
               onChange={resolveComment}
             />
+            <Button size="sm" onClick={deleteComment}>
+              Delete
+            </Button>
           </td>
         ) : null}
       </tr>
```
Comment (dleard) [inline app/containers/Applications/ApplicationCommentsByForm.tsx]: We are displaying createdAt below the comment here. We should also show the name of the user who created the column (like Hamza's XD). There is a ciipUserByCreatedBy foreign key connection in reviewComment that can be used to get the user's name.

File: app/containers/Applications/ApplicationCommentsContainer.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ export const ApplicationCommentsComponent: React.FunctionComponent<Props> = prop
               <thead style={{textAlign: 'center'}}>
```
Comment (dleard) [inline app/containers/Applications/ApplicationCommentsContainer.tsx]: Each of the two comment tables should probably have some sort of distinct title describing why the comments are separated, like 'General Comments' & 'Requested Changes' or something like that.

File: app/mutations/application/deleteReviewCommentMutation.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    const formResultRoot = store.get(formResultId);
+    const commentConnection = ConnectionHandler.getConnection(
+      formResultRoot,
+      connectionKey
+    );
+    ConnectionHandler.deleteNode(commentConnection, variables.input.id);
+  };
+
+  const m = new BaseMutation<updateReviewCommentMutationType>(
+    'update-review-comment-mutation'
```
Comment (dleard) [inline app/mutations/application/deleteReviewCommentMutation.ts]:
```suggestion
    'delete-review-comment-mutation'
```

Comment (dimak1) [reply app/mutations/application/deleteReviewCommentMutation.ts]: Updated

File: app/containers/Applications/ApplicationCommentsContainer.tsx (hunk 2 above)
Comment (dimak1) [reply app/containers/Applications/ApplicationCommentsContainer.tsx]: Added titles for columns

File: app/containers/Applications/ApplicationCommentsByForm.tsx (hunk 1 above)
Comment (dimak1) [reply app/containers/Applications/ApplicationCommentsByForm.tsx]: Added user name below the comment

**Comment by dleard** [comment]
closes #338

File: app/containers/Applications/ApplicationCommentsByForm.tsx [hunk 4]
```diff
@@ -35,25 +40,92 @@ export const ApplicationCommentsByForm: React.FunctionComponent<Props> = props =
...
   return (
     <>
       <tr>
-        <td>{reviewComment.description}</td>
+        <td>
+          <div className="description">{reviewComment.description}</div>
+          <div>
+            <small style={{color: '#777'}}>
+              {reviewComment.ciipUserByCreatedBy.firstName}{' '}
+              {reviewComment.ciipUserByCreatedBy.firstName} (
```
Comment (dleard) [inline app/containers/Applications/ApplicationCommentsByForm.tsx]: should this be 
```suggestion
              {reviewComment.ciipUserByCreatedBy.lastName} (
```
?

Comment (dimak1) [reply app/containers/Applications/ApplicationCommentsByForm.tsx]: Ohh good catch, fixed

**Pull request merged**